### PR TITLE
Add auth filter

### PR DESCRIPTION
**Pull request by kaka89**
Add auth filter for check auth before send calls to back end services.

### CODE REVIEW

File: build_scripts/application_dependencies.gradle [hunk 1]
```diff
@@ -18,12 +18,6 @@ dependencies {
     // For @CreatedDate and @LastModifiedDate
     compile('com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.8.5')
 
-    // Spring data jpa
-    compile('org.springframework.boot:spring-boot-starter-data-jpa')
-    // Postgresql driver
-    compile('org.postgresql:postgresql:9.4.1212')
-
-
     compile('com.google.api-client:google-api-client:1.22.0')
```
Comment (davis999) [inline build_scripts/application_dependencies.gradle]: remove useless dependency.

File: src/main/java/io/reactivesw/gateway/config/AuthFilterConfig.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ */
+@Configuration
+@ConfigurationProperties(prefix = "auth.ignored")
+@Data
+public class AuthFilterConfig {
+
+  /**
+   * Ignored hosts that do not need to check auth.
+   */
+  private List<String> hosts = new ArrayList<>();
```
Comment (davis999) [inline src/main/java/io/reactivesw/gateway/config/AuthFilterConfig.java:21]: when and how to inject value?
missing annotation?

Comment (kaka89) [reply src/main/java/io/reactivesw/gateway/config/AuthFilterConfig.java:21]: on class header

File: build_scripts/application_dependencies.gradle (hunk 1 above)
Comment (kaka89) [reply build_scripts/application_dependencies.gradle]: done.

**Pull request merged**